### PR TITLE
Feat: storybook 7 migration 🎉

### LINKED ISSUE #12912: Storybook: Migrate to v7
- What do we get out of this upgrade?
- How important is it to our users?
- What do we need to do to prepare for it, if anything?

https://github.com/carbon-design-system/carbon/issues/12912#tasklist-block-0a692a62-ff35-4e76-ac55-73c74bfe6145

### LINKED ISSUE #13379: Storybook 7: Explore update
Storybook v7 [has been released](https://github.com/storybookjs/storybook/releases/tag/v7.0.2). This issue is for updating the monorepo to v7. We should defer as many optional migrations as possible so we can get on v7 as quick as possible and then iterate from there.

```[tasklist]
- [x] View and use the [migration guide](https://storybook.js.org/docs/react/migration-guide#page-top)
- [x] Run the migrator script from the docs
- [x] Fix/report/mitigate any issues
- [x] Can we defer mdx2 update? Is mdx2 update large?
- [x] Remove reactOptions, legacyRootApi: false in main.js
```

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #13379 
Closes #12912

Upgrades Storybook to v7 and adds all necessary modifications/changes to ensure parity with current prod deployment

#### Changelog

**New**

- .babelrc.json config file at top level of react package, removed from react/.storybook

**Changed**

- Stories now have a `Overview` docs page on the left side nav instead of a "Docs" tab
- now using MDX2 for .mdx docs
- changed storybook build command in workflow files

**Removed**

- Welcome page docs, now only using welcome page as a landing page
- packages/react/.storybook/.babelrc.js, in favor of packages/react/.babelrc.json
- [story].ArgTypes.defaultValue in favor of [story].Args
- @storybook/addon-docs mdx imports in favor of @storybook/blocks

#### Testing / Reviewing

`yarn install`
`yarn build`
`yarn run storybook`

Please test all things storybook 🙏🏻, docs, playground controls, skeleton, default stories, feature flags, actions... Should all be running smoothly and let me know of any issues 

#### NOTE

it is expected for some of the CIs to fail since the command used to build storybook changed in V7 and `main` branch still doesn't know about it. you'll need to build this manually and test it for this PR :)

### CODE REVIEW

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
This looks great!! 🥳  Only noticed a couple super minor things posted below. 😄 

Button overview
Screenshot 2023-06-13 at 9 45 54 AM

Composed modal overview, 7 on the left, live site on the right. This might be ok? 
Screenshot 2023-06-13 at 9 52 28 AM


Slide overview, example is disabled, differs from what is on live.
Screenshot 2023-06-13 at 10 14 36 AM

Is there a way to have overview first here? 
Screenshot 2023-06-13 at 10 19 18 AM

**Comment by alisonjoseph** [comment]
This is unrelated to this PR, but can we add padding here? I can also create a separate issue/PR for this. 
Screenshot 2023-06-13 at 10 09 57 AM

**Comment by francinelucca** [comment]
@alisonjoseph nice catches!

- Button Overview docs should be fixed
- Added padding to OrderedList docs
- Overview should be showing up first in all stories now

- The composedModal notification changes I did on purpose, I was modeling after the notifications in https://react.carbondesignsystem.com/?path=/docs/components-modal--default. I thought the ack of a CodeSnippet in ComposedModal might have been an oversight 🤔. I can revert back to just quotes but I think they should be the same in both places
- For the Slider, is it possible that you went into the Playground, turned on the disabled control and then went into the overview and that's when you saw it disabled? I don't see it disabled on first render on my end.

Let me know if you find anything else 🙏🏻, TIA!

**Review by andreancardona — APPROVED** [review]
Wow 🔥 I feel like I did a pretty thorough review and caught some bugs on our end. This is more a question - I assume it's a v7 feature - but just saw a small discrepancy with the `deprecated` props in the docs. 
Screenshot 2023-06-19 at 21 31 46
Screenshot 2023-06-19 at 21 32 04

**Review by alisonjoseph — APPROVED** [review]
This all LGTM me now! I think it would be nice to add some styling to the new deprecated messages that show up in the props table. We can just make a new issue for design for that and add it in a future PR, don't think it needs to block this. 🥳

**Review by tay1orjones — APPROVED** [review]
This update looks super good! I pushed a commit to update the links to the feature flag overview page. That page URL changed slightly with the update.

> it is expected for some of the CIs to fail since the command used to build storybook changed in V7 and main branch still doesn't know about it. you'll need to build this manually and test it for this PR :)

Once this merges I'll go in and update the Netlify config settings so the deploy previews will work moving forward 👍

**Comment by francinelucca** [comment]
> but just saw a small discrepancy with the `deprecated` props in the docs.

Yeah I believe that's a consequence of switching from `<Props />` to `<ArgsTable />`. I created an issue to have design take a look as per @alisonjoseph comment https://github.com/carbon-design-system/carbon/issues/14040

File: packages/react/.storybook/main.js [hunk 1]
```diff
@@ -7,26 +7,36 @@
 
 'use strict';
 
+import remarkGfm from 'remark-gfm';
 const fs = require('fs');
 const glob = require('fast-glob');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const path = require('path');
 
+const storyGlobs =
```
Comment (tay1orjones) [inline packages/react/.storybook/main.js:18]: Could you add a brief comment here explaining why we conditionally exclude some globs when this flag is false?

**Review by tay1orjones — APPROVED** [review]
Yesss, looks great! Lets go! 🎉

**Pull request merged**